staging/src/k8s.io/apiserver/pkg/util/flowcontrol/rule.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	rmtypesv1alpha1 "k8s.io/api/flowcontrol/v1alpha1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+)
+
+func matchesFlowSchema(digest RequestDigest, flowSchema *rmtypesv1alpha1.FlowSchema) bool {
+	for _, policyRule := range flowSchema.Spec.Rules {
+		subjectMatches := false
+		for _, subject := range policyRule.Subjects {
+			if matchesSubject(digest, subject) {
+				subjectMatches = true
+				break
+			}
+		}
+		if !subjectMatches {
+			continue
+		}
+
+		if matchesPolicyRule(digest, policyRule.Rule) {
+			return true
+		}
+	}
+	return false
+}
+
+func matchesSubject(digest RequestDigest, subject rmtypesv1alpha1.Subject) bool {
+	user := digest.User
+	switch subject.Kind {
+	case rmtypesv1alpha1.UserKind:
+		return user.GetName() == subject.Name
+	case rmtypesv1alpha1.GroupKind:
+		return containsString(subject.APIGroup, user.GetGroups(), "")
+	case rmtypesv1alpha1.ServiceAccountKind:
+		return serviceaccount.MatchesUsername(subject.Namespace, subject.Name, user.GetName())
+	default:
+		return false
+	}
+}
+
+func matchesPolicyRule(digest RequestDigest, policyRule rmtypesv1alpha1.PolicyRule) bool {
+	if !matchPolicyRuleVerb(policyRule.Verbs, digest.RequestInfo.Verb) {
```

Comment:
Why not make this line
```
    if !containsString(digest.RequestInfo.Verb, policyRule.Verbs, rmtypesv1alpha1.VerbAll) {
```
?

Note that `policyRule.Verbs` is declared to have type `[]string`, so going directly to a function of `[]string` does not break any abstraction that exists in the API.

Reply:
am in favor of a named function for readability